src/test/scala/com/ambiata/poacher/mr/ArgsSpec.scala

```diff
@@ -0,0 +1,34 @@
+package com.ambiata.poacher.mr
+
+import org.scalacheck._
+import org.specs2._
+
+class ArgsSpec extends Specification with ScalaCheck { def is = s2"""
+
+  Parse args and create configuration           $configuration
+"""
+
+  def configuration = prop { (a1: List[String], ha: HadoopArgs) =>
+    val (c, a2) = Args.configuration(ha.args.map(_.arg) ++ a1)
```

Comment:
I would like to add more naming to those variables and separate the 2 behaviours (maybe in 2 properties):
```
val (configuration, remainingArgs) = Args.configuration(ha.args.map(_.arg) ++ a1)
```
and 
```
val parsedValues = ha.args.map(a => configuration.get(a.k))
val expectedValues = ha.args.map(_.v)

(remainingArgs === a1) and
(parsedValues == expectedValues)
```